src/main/java/com/google/sps/data/RentPost.java

```diff
@@ -0,0 +1,152 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.CURRENCY_CODE;
+import static com.google.sps.Constants.DATE_FORMAT;
+import static com.google.sps.Constants.RENT_COLLECTION_NAME;
+import static com.google.sps.Constants.REQUEST_DESCRIPTION;
+import static com.google.sps.Constants.REQUEST_END_DATE;
+import static com.google.sps.Constants.REQUEST_LEASE_TYPE;
+import static com.google.sps.Constants.REQUEST_NUM_ROOMS;
+import static com.google.sps.Constants.REQUEST_PRICE;
+import static com.google.sps.Constants.REQUEST_ROOM_TYPE;
+import static com.google.sps.Constants.REQUEST_START_DATE;
+import static com.google.sps.Constants.REQUEST_TIMESTAMP;
+import static com.google.sps.Constants.REQUEST_TITLE;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import org.javamoney.moneta.Money;
+
+/** A rent post made by a user. */
+@AutoValue
+public abstract class RentPost {
+  enum LeaseType {
+    YEAR_LONG, 
+    MONTH_TO_MONTH, 
+    SUMMER_SUBLET;
+  }
+  enum RoomType {
+    SINGLE,
+    SHARED;
+  }
+
+  public abstract String description();
+  @Nullable public abstract Date endDate();
+  public abstract LeaseType leaseType();
+  public abstract int numRooms();
+  public abstract Money price();
+  public abstract RoomType roomType();
```

Comment:
So to be clear, the lister is supposed to just make multiple listings if these properties don't all match? So for instance if I have two singles and one double available, I would make one listing for the two singles and one listing for the double? Seems like there should be some way to indicate how many people are sharing the house/apartment total, and I don't think that's captured here. 

Side note: what about bathrooms? That's kind of a big deal too, particularly regarding how many people are sharing (goes to my point above)

Reply:
Added the below fields:

- number of bathrooms
- number of single bedrooms available for rent
- number of shared bedrooms available for rent
- total number of rooms in the house
- cost of a single bedroom
- cost of a shared bedroom
- total cost of listing (calculated by the sum over all room types of the cost of room type * num of rooms of that type)